File: typescript/src/api/resources/runs/client/requests/StartHarnessRunRequest.ts

import * as Sikaru from "../../../../index.js";

/**
 * @example
 *     {
 *         input: {
 *             "key": "value"
 *         },
 *         policy: {
 *             "key": "value"
 *         },
 *         product_context: {
 *             "key": "value"
 *         },
 *         tenant_id: "tenant_id",
 *         user_id: "user_id"
 *     }
 */
export interface StartHarnessRunRequest {
    capability_grants?: string[];
    compute_provider_id?: string | null;
    conversation_id?: string | null;
    correlation_id?: string | null;
    event_delivery?: Sikaru.EventDeliveryRequest;
    execution_environment?: StartHarnessRunRequest.ExecutionEnvironment;
    input: Record<string, unknown>;
    job_id?: string | null;
    policy: Record<string, unknown>;
    product_context: Record<string, unknown>;
    run_mode?: StartHarnessRunRequest.RunMode;
    tenant_id: string;
    tool_provider_refs?: Sikaru.ToolProviderRefRequest[];
    trace_id?: string | null;
    user_id: string;
}

export namespace StartHarnessRunRequest {
    export const ExecutionEnvironment = {
            Managed: "managed",
            Local: "local"
        } as const;
    export type ExecutionEnvironment = typeof ExecutionEnvironment[keyof typeof ExecutionEnvironment];
    export const RunMode = {
            Interactive: "interactive",
            BackgroundSubagent: "background_subagent",
            SandboxExecution: "sandbox_execution"
        } as const;
    export type RunMode = typeof RunMode[keyof typeof RunMode];
}

File: ruby/lib/Sikaru/runs/types/start_harness_run_request.rb
# frozen_string_literal: true

module Sikaru
  module Runs
    module Types
      class StartHarnessRunRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :harness_id, -> { String }, optional: false, nullable: false

        field :auto_improve, -> { Internal::Types::Boolean }, optional: true, nullable: false

        field :capability_grants, -> { Internal::Types::Array[String] }, optional: true, nullable: false

        field :compute_environment_id, -> { String }, optional: true, nullable: false

        field :compute_workspace_provenance, -> { Sikaru::Types::WorkspaceProvenance }, optional: true, nullable: false

        field :conversation_id, -> { String }, optional: true, nullable: false

        field :correlation_id, -> { String }, optional: true, nullable: false

        field :event_delivery, -> { Sikaru::Types::EventDeliveryRequest }, optional: true, nullable: false

        field :idempotency_key, -> { String }, optional: true, nullable: false

        field :input, -> { Internal::Types::Hash[String, Object] }, optional: false, nullable: false

        field :job_id, -> { String }, optional: true, nullable: false

        field :model, -> { String }, optional: true, nullable: false

        field :policy, -> { Internal::Types::Hash[String, Object] }, optional: false, nullable: false

        field :product_context, -> { Internal::Types::Hash[String, Object] }, optional: false, nullable: false

        field :run_mode, -> { Sikaru::Runs::Types::StartHarnessRunRequestRunMode }, optional: true, nullable: false

        field :tenant_id, -> { String }, optional: false, nullable: false

        field :tool_provider_refs, -> { Internal::Types::Array[Sikaru::Types::ToolProviderRefRequest] }, optional: true, nullable: false

        field :trace_id, -> { String }, optional: true, nullable: false

        field :user_id, -> { String }, optional: false, nullable: false
      end
    end
  end
end
